thor/bytes32.go

```diff
@@ -46,9 +45,7 @@ func (b Bytes32) IsZero() bool {
 
 // MarshalJSON implements json.Marshaler.
 func (b Bytes32) MarshalJSON() ([]byte, error) {
-	// Convert Bytes32 to a hexadecimal string.
-	// if []byte = [00000...] then we return 0x000000 instead of nil
-	return json.Marshal(hexutil.Encode(b[:]))
+	return json.Marshal(b.String())
```

Comment:
Perhaps redundant, but after chasing my tail for a while this would have helped.

```suggestion
	// Bytes32 are represented as a hexadecimal string.
	return json.Marshal(b.String())
```